core/client/src/main/java/alluxio/client/block/BlockStoreContext.java

```diff
@@ -252,33 +177,20 @@ private BlockWorkerClient acquireRemoteWorkerClient(WorkerNetAddress address) {
    *        this method is called
    */
   public void releaseWorkerClient(BlockWorkerClient blockWorkerClient) {
-    // If the client is local and the pool exists, release the client to the pool, otherwise just
-    // close the client.
-    if (blockWorkerClient.isLocal()) {
-      // Return local worker client to its resource pool.
-      WorkerNetAddress address = blockWorkerClient.getWorkerNetAddress();
-      if (!mLocalBlockWorkerClientPoolMap.containsKey(address)) {
-        LOG.error("The client to worker at {} to release is no longer registered in the context.",
-            address);
-        blockWorkerClient.close();
-      } else {
-        mLocalBlockWorkerClientPoolMap.get(address).release(blockWorkerClient);
-      }
-    } else {
-      // Destroy remote worker client.
-      blockWorkerClient.close();
-    }
+    Preconditions.checkNotNull(blockWorkerClient);
+    blockWorkerClient.close();
   }
 
   /**
-   * Acquire a netty channel from the channel pools.
+   * Acquires a netty channel from the channel pools.
```

Comment:
make javadoc more complete / accurate? currently it is not clear from the doc what will happen if the address is not in the pool. Actually how address is used is not javadoced.